source/s3_request_messages.c

```diff
@@ -457,8 +458,37 @@ static const struct aws_byte_cursor s_slash_char = AWS_BYTE_CUR_INIT_FROM_STRING
  */
 struct aws_http_message *aws_s3_get_source_object_size_message_new(
     struct aws_allocator *allocator,
-    struct aws_http_message *base_message) {
+    struct aws_http_message *base_message,
+    struct aws_uri *source_uri) {
     struct aws_http_message *message = NULL;
+
+    message = aws_http_message_new_request(allocator);
```

Comment:
Lot of the error handling down there assumes there was nothing to clean up and just returns null.

Need to update them since we now allocate the message at the beginning of the function.